onnxruntime/core/platform/windows/debug_alloc.cc

```diff
@@ -75,45 +75,43 @@ struct SymbolHelper {
 
   SymbolHelper() = default;
 
-  static constexpr size_t kInitialBufferSize = sizeof(SYMBOL_INFO) + MAX_SYM_NAME;
+  static constexpr size_t kSymbolBufferSize = sizeof(SYMBOL_INFO) + MAX_SYM_NAME;
 
-  bool LoookupSymAndInitialize(const ULONG_PTR address, char* buffer, size_t buffer_size, SYMBOL_INFO* symbol) {
-    if (SymFromAddr(process_handle_, address, 0, symbol) != TRUE) {
+  bool LookupSymAndInitialize(const void* address, SYMBOL_INFO* symbol, std::ostream& message) {
+    if (SymFromAddr(process_handle_, reinterpret_cast<ULONG_PTR>(address), 0, symbol) != TRUE) {
       if (GetLastError() == ERROR_INVALID_HANDLE) {
         // Try to initialize first
-        if (!InitializeWhenNeeded() || SymFromAddr(process_handle_, address, 0, symbol) != TRUE) {
-          _snprintf_s(buffer, buffer_size, _TRUNCATE, "0x%08IX (Unknown symbol)", address);
+        if (!InitializeWhenNeeded() ||
+            SymFromAddr(process_handle_, reinterpret_cast<ULONG_PTR>(address), 0, symbol) != TRUE) {
+          message << "0x" << address << " (Unknown symbol)";
           return false;
         }
       } else {
-        _snprintf_s(buffer, buffer_size, _TRUNCATE, "0x%08IX (Unknown symbol)", address);
+        message << "0x" << address << " (Unknown symbol)";
         return false;
       }
     }
     return true;
   }
 
-  void Lookup(std::string& string, const ULONG_PTR address) {
-    alignas(SYMBOL_INFO) char buffer[kInitialBufferSize] = {0};
-    SYMBOL_INFO* symbol = reinterpret_cast<SYMBOL_INFO*>(buffer);
+  void Lookup(const void* address, std::ostream& message) {
+    alignas(SYMBOL_INFO) char symbol_buffer[kSymbolBufferSize] = {0};
```

Comment:
```suggestion
    alignas(SYMBOL_INFO) std::byte symbol_buffer[kSymbolBufferSize] = {0};
```
To avoid anyone accidentally writing directly to symbol_buffer as a string again?

Reply:
I found `SYMBOL_INFO_PACKAGE` in the DbgHelp.h, which is a struct containing `SYMBOL_INFO` and a char buffer following it. using that instead.

Reply:
Is that buffer owned by the `SYMBOL_INFO_PACKAGE`?

Reply:
yes. here's the definition in DbgHelp.h:

```c++
typedef struct _SYMBOL_INFO_PACKAGE {
    SYMBOL_INFO si;
    CHAR        name[MAX_SYM_NAME + 1];
} SYMBOL_INFO_PACKAGE, *PSYMBOL_INFO_PACKAGE;
```